client.go

```diff
@@ -267,8 +275,13 @@ func (c *Client) nextID() uint32 {
 func (c *Client) recvVersion() error {
 	typ, data, err := c.recvPacket(0)
 	if err != nil {
+		if err == io.EOF {
+			return io.ErrUnexpectedEOF
```

Comment:
Maybe we could do something like this here

```
return fmt.Errorf("server unexpectedly closed connection: %w", io.ErrUnexpectedEOF)
```

so `errors.Is(err, io.ErrUnexpectedEOF)` will work